tests/acceptance/bootstrap/AuthAppContext.php

```diff
@@ -101,4 +101,54 @@ public function userListsAllCreatedTokensUsingTheAuthAppApi(string $user): void
 			)
 		);
 	}
+
+	/**
+	 * @Given the administrator has created app token for user :impersonatedUser with expiration time :expiration via Impersonation API
+	 *
+	 * @param string $impersonatedUser
+	 * @param string $expiration
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasCreatedAppTokenWithExpirationTimeImpersonatingUserUsingTheApi(
+		string $impersonatedUser,
+		string $expiration,
+	): void {
+		$response = AuthAppHelper::createAuthAppTokenViaImpersonationAPI(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getAdminUsername(),
+			$this->featureContext->getAdminPassword(),
+			$expiration,
+			$this->featureContext->getActualUsername($impersonatedUser),
+		);
+		$this->featureContext->theHTTPStatusCodeShouldBe(
+			200,
+			"Failed creating auth-app token\n"
+			. "HTTP status code 200 is not the expected value " . $response->getStatusCode(),
+			$response
+		);
+	}
+
+	/**
+	 * @When the administrator creates app token for user :impersonatedUser with expiration time :expiration via impersonation API
```

Comment:
this is not a different api.
```suggestion
	 * @When the administrator creates app token for user :impersonatedUser with expiration time :expiration using the auth-app API
```